src/main/java/org/broadinstitute/hellbender/tools/exome/AllelicCountCollection.java

```diff
@@ -30,24 +33,29 @@ public AllelicCountCollection() {
     }
 
     /**
-     * Constructor that reads (sequence, position, reference count, alternate count) from the specified file.
-     * @param inputFile     file to read from
+     * Constructor from from file. Checks whether the input file has just the basic columns, or full columns.
+     * @param inputFile file to read from
      */
     public AllelicCountCollection(final File inputFile) {
         Utils.nonNull(inputFile);
         Utils.regularReadableUserFile(inputFile);
+
         try (final TableReader<AllelicCount> reader = TableUtils.reader(inputFile,
                 (columns, formatExceptionFactory) -> {
-                    TableUtils.checkMandatoryColumns(columns, AllelicCountTableColumns.COLUMN_NAME_ARRAY, formatExceptionFactory);
-
-                    // return the lambda to translate dataLines into AllelicCounts.
-                    return (dataLine) -> {
-                        final int position = dataLine.getInt(AllelicCountTableColumns.POSITION.toString());
-                        final SimpleInterval interval = new SimpleInterval(dataLine.get(AllelicCountTableColumns.CONTIG.toString()), position, position);
-                        final int refReadCount = dataLine.getInt(AllelicCountTableColumns.REF_COUNT.toString());
-                        final int altReadCount = dataLine.getInt(AllelicCountTableColumns.ALT_COUNT.toString());
-                        return new AllelicCount(interval, refReadCount, altReadCount);
-                    };
+                    /* in the order of decrasing verbosity */
```

Comment:
Since all this ifs are actually mutually exclusive (due to the return statements) it makes more sense to do some if-else-if-else-... -else instead; is rather more neat:

```
if (...) {
   return A;
} else if (...) {
   return B;
} else if (...
   ...
} else {
   throw X;
}
```